ocs_ci/ocs/resources/cloud_manager.py

```diff
@@ -0,0 +1,305 @@
+import base64
+import logging
+import os
+from abc import ABC, abstractmethod
+import boto3
+from botocore.exceptions import ClientError
+from time import sleep
+from google.cloud import storage
+from google.auth.exceptions import DefaultCredentialsError
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.utility import templating
+from ocs_ci.utility.utils import TimeoutSampler
+from tests.helpers import create_resource
+
+logger = logging.getLogger(name=__file__)
+
+
+class CloudManager(ABC):
+    """
+    Class containing all client types
+
+    """
+    aws_client, google_client, azure_client, s3comp_client = (None,) * 4
+
+    def __init__(self):
+        # TODO: solve credentials for clients (working with local creds for now)
+        self.aws_client = S3Client()
+        # TODO Need credentials to check
+        self.google_client = None
+        self.azure_client = None
+        self.s3comp_client = None
+
+
+class CloudClient(ABC):
+    """
+    Base abstract class for Cloud based API calls
+
+    """
+    client = None
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def create_uls(self, name, region):
+        """
+        Super method that first logs the Underlying Storage creation and then calls
+        the appropriate implementation
+
+        """
+        logger.info(f"Creating Underlying Storage {name} in {region}")
```

Comment:
Please check whether it's possible to also print out the client type used

Reply:
added an info print to identify the cloud in the uls factory